modules/Bio/EnsEMBL/Compara/DBSQL/AlignSliceAdaptor.pm

```diff
@@ -433,111 +381,101 @@ sub _combine_genomic_align_trees {
 
     my $match = 0;
     ## SECONDARY LOOP. Note that the $species_counter is not reset at the end of the loop.
-    ## This ensures that we do not add two nodes to the same species_order track and that we
-    ## preserve the order in all existing and in the new tree.
-    while (!$match and $species_counter < @$species_order) {
-      my $species_genome_db = $species_order->[$species_counter]->{genome_db};
-      my $species_right_node_id = $species_order->[$species_counter]->{right_node_id};
+    ## This ensures that we do not add two nodes to $species_order and that we preserve as
+    ## far as possible the order in all existing and in the new trees.
+    while (!$match) {
+      my $species = $species_order->[$species_counter];
+      my ($species_genome_db, $species_right_node_id);
+      $species_genome_db = $species->{genome_db} if defined $species;
+      $species_right_node_id = $species->{right_node_id} if defined $species;
       $match = 1;
 
-      ## 1. Use info from species_right_node_id if available
+      ## 1. Use info from $species_right_node_id if available
       if (defined($species_right_node_id) and $species_right_node_id == $this_node_id) {
-          $species_order->[$species_counter]->{right_node_id} = $this_right_node_id;
-          # #         $species_order->[$species_counter]->{last_node} = $this_genomic_align_node;
+        $species_order->[$species_counter]->{right_node_id} = $this_right_node_id;
         push (@{$species_order->[$species_counter]->{genomic_align_ids}}, @$these_genomic_align_ids);
+        if (defined $pending_ancestral_species) {
+          my $prev_species_index = $species_counter - 1;
+          $species_order->[$prev_species_index]->{right_node_id} = $pending_ancestral_species->{right_node_id};
+          push @{$species_order->[$prev_species_index]->{genomic_align_ids}}, @{$pending_ancestral_species->{genomic_align_ids}};
+          undef $pending_ancestral_species;
+        }
         ## DEBUG info
```

Comment:
```suggestion
```
May as well delete this too, since `_debug_info` as sub name is pretty self explanatory.